stdlib/compat.ncl

```diff
@@ -0,0 +1,12 @@
+{
+compat | doc "Nix compatibility layer. should not be used by Nickel program."
+= {
+  with: Array {_: Dyn} -> Str -> Dyn
```

Comment:
I don't know if we want a type annotation right now. In spirit, this is great. In practice, it incurs a contract check each time. Maybe just typecheck it once and then remove it? I guess it could be nice to have pragmas to say "don't generate contract application for this file", but we don't have that yet